pkg/osbuild/rhsm_stage.go

```diff
@@ -52,3 +57,55 @@ type SubManConfigRHSMCERTDSection struct {
 	// Automatic system registration
 	AutoRegistration *bool `json:"auto_registration,omitempty"`
 }
+
+func NewRHSMStageOptions(config *subscription.RHSMConfig) *RHSMStageOptions {
```

Comment:
A bit of a meta question, we are doing a lot of work here convert and also deal with pointers - what if we would just make those normal structs without points so that clone would work naturally and then detect what sub-part is enabled/not-enabled by some other mechanism? Maybe a bad/unworkable idea but seeing all the scafolding here makes me wonder :)

Reply:
The only reason why we use pointers to `bool` is that we actually need to distinguish if the (`false`) value was explicitly set in the BP / internal image configuration vs. being the default type value that Go initialized it to. Otherwise, we could be unintentionally disabling things in the configuration by the osbuild stage.

We need to use pointers to struct in the osbuild stage options struct to not have them serialized if marked as `omitempty`.

But your comment made me think about this more. We probably don't need to use pointers to structs, only to `bool`, in the internal RHSMConfig representation. I'll try to rework it.

Reply:
I reduced the amount of pointers to the bare minimum - only to the primitive types, which simplified the code a lot.

Reply:
Thank you! Really appreciated